dm/worker/server.go

```diff
@@ -493,19 +659,71 @@ func (s *Server) operateSourceBound(bound ha.SourceBound) error {
 	if !ok {
 		return terror.ErrWorkerFailToGetSourceConfigFromEtcd.Generate(bound.Source)
 	}
-	w, err := s.getOrStartWorker(&sourceCfg)
+	w, err := s.getOrStartWorker(&sourceCfg, true)
+	s.setSourceStatus(bound.Source, err, true)
 	if err != nil {
 		return err
 	}
-	if sourceCfg.EnableRelay {
-		s.UpdateKeepAliveTTL(s.cfg.RelayKeepAliveTTL)
-		if err = w.EnableRelay(); err != nil {
-			return err
-		}
-	}
 	return w.EnableHandleSubtasks()
```

Comment:
Should we `setSourceStatus` when `w.EnableHandleSubtasks()` returns error?

Reply:
now set in caller. it's a good idea to move `setSourceStatus ` into this function, and I prefer to change it in next PR

Reply:
oh rebase is a huge problem